src/cpp/core/system/PosixChildProcess.cpp

```diff
@@ -906,6 +927,7 @@ void AsyncChildProcess::poll()
    pAsyncImpl_->initTimers(options().reportHasSubprocs);
    if (pAsyncImpl_->checkChildCount())
    {
+      pAsyncImpl_->hasRecentOutput_ = false;
```

Comment:
`hasRecentOutput_` gets flipped to false every second (that's the frequency of `checkChildCount` being true, and is set to true whenever there is output seen in the terminal polling loop. The value of hasRecentOutput_ is checked (or'd with the child-process flag) here: https://github.com/rstudio/rstudio/blob/88b6ae2139cae85bcdb3f9aa03219e9285ca3305/src/cpp/session/SessionHttpMethods.cpp#L401